packages/stencil/src/components/datacom-overflow-menu/datacom-overflow-menu.tsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Component,
+  Host,
+  h,
+  Prop,
+  State,
+  Element,
+  Listen,
+} from '@stencil/core';
+import { OverflowMenuIcon } from './assets/overflow-menu-icon';
+
+export type MenuSizeType = 'standard' | 'small';
+export type OverFlowMenuVariantType = 'horizontal' | 'vertical';
+export type MenuItemsPositionType = 'center' | 'left' | 'right';
+
+@Component({
+  tag: 'datacom-overflow-menu',
+  styleUrl: 'datacom-overflow-menu.css',
+  shadow: true,
+})
+export class DatacomOverflowMenu {
+  @Prop() variant: OverFlowMenuVariantType = 'horizontal';
+  @Prop() label: string;
+  @State() isOpen: boolean = false;
+  @Prop() size: MenuSizeType = 'small';
+  @Element() hostElement: HTMLDatacomOverflowMenuElement;
+  @Prop({ mutable: true }) position: MenuItemsPositionType = 'center';
+  private buttonRef: HTMLButtonElement;
+  private firstElementRef: HTMLElement;
+
+  private open() {
+    this.isOpen = true;
+  }
+  private close() {
+    this.isOpen = false;
+  }
+
+  handleKeyUp = (event: KeyboardEvent) => {
+    if (event.key === 'Enter' || event.key === 'Return' || event.key == ' ') {
+      this.open();
+      setTimeout(() => {
+        this.setFocusToFirstItem();
+      }, 100);
+      return;
+    }
+    if (event.key === 'Escape' && this.isOpen) {
+      this.close();
+      this.buttonRef.focus();
+      return;
+    }
+  };
+  /**
+   *Set the focus first element of dropdown options when click on the menu button
+   *First child of slot
+   */
+  private setFocusToFirstItem() {
+    this.firstElementRef = this.hostElement.shadowRoot
+      .querySelector('.dc-overflow-dropdown-options')
+      ?.querySelector('slot')
+      .assignedElements()[0]
+      .shadowRoot.querySelector('.dc-menu-item-wrapper') as HTMLElement;
+    if (this.firstElementRef) {
+      this.firstElementRef.setAttribute('tabindex', '0');
+      this.firstElementRef.focus();
+    }
+  }
+
+  private toggleMenu = (event: MouseEvent) => {
+    event.preventDefault();
+    if (this.isOpen !== true) {
+      this.open();
+      setTimeout(() => {
+        const sampleText =
```

Comment:
Please change to a proper constant naming. e.g element

Reply:
Done